### PR TITLE
Defines how quantity of products that have unitMultiplier will work

### PR DESCRIPTION
**Pull request by germanBonacchi**
#### What problem is this solving?

When the product have unitMultiplier, if you add more items, the quantity will be affected with the unitMultiplier.
With this prop the quantity will be not affected with the unitMultiplier.
This was required for a client who want to show in the pdp and the gallety, the quantity of products that have unitMultiplier as units.

#### How to test it?

Changed the product quantity on pdp and gallery.

[Workspace] https://fpatest01--arvitalb2bqa.myvtex.com/

#### Screenshots or example usage:

Grabación de pantalla 2021-06-01 a la(s) 17 08 59 (1)

#### Describe alternatives you've considered, if any.

.

### CODE REVIEW

**Comment by victorhmp** [comment]
Hi @germanBonacchi!
The comments I made at https://github.com/vtex-apps/product-list/pull/103 also apply here :)

**Comment by germanBonacchi** [comment]
i did the changes that you comment in both PR's

File: CHANGELOG.md
Comment (victorhmp) [inline CHANGELOG.md:11]: Please update this entry

File: docs/README.md
Comment (victorhmp) [inline docs/README.md:69]: I don't think this description illustrates how this prop is working now. Could you also update it?

File: manifest.json
Comment (victorhmp) [inline manifest.json:4]: No need to change this, the CI/CD bot will handle the version bump once the PR is merged :)

```suggestion
  "version": "1.6.0",
```

File: react/components/BaseProductQuantity.tsx
Comment (victorhmp) [inline react/components/BaseProductQuantity.tsx:68]: This could be simplified into something like:

```suggestion
  const unitMultiplier = quantitySelectorStep === 'singleUnit' ? 1 : selectedItem.unitMultiplier
  
```

Comment (victorhmp) [reply react/components/BaseProductQuantity.tsx:68]: The lint will probably complain that this line is too long, so I would recommend typing it out in your editor and letting ESLint/Prettier do the magic

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: docs/README.md
Comment (germanBonacchi) [reply docs/README.md:69]: Do you think that this ilustrate more? 
| `quantitySelectorStep` | `enum` | Show as single units the quantity of products that have unitMultiplier. Possible values are: `singleUnit` (the quantity will be affected with the unitMultiplier) and `unitMultiplier` (the quantity will be not affected with the unitMultiplier). |

File: docs/README.md
Comment (igorbrasileiro) [inline docs/README.md:69]:
```suggestion
| `quantitySelectorStep` | `enum` | Defines how the number of products that have unitMultiplier will show. Possible values are: `singleUnit` (the quantity will be not affected with the unitMultiplier) and `unitMultiplier` (the quantity will be affected with the unitMultiplier). | `unitMultiplier` |
```

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Just a small fix on doc

File: CHANGELOG.md
Comment (igorbrasileiro) [inline CHANGELOG.md:11]:
```suggestion
- `quantitySelectorStep` prop to `product-quantity` and `product-summary-quantity`. Enables users to see the number of products affected with the unitMultiplier like units.
```

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]

**Review by igorbrasileiro — APPROVED** [review]

**Comment by germanBonacchi** [comment]
@victorhmp Is everything ready to approve?

File: CHANGELOG.md
Comment (victorhmp) [inline CHANGELOG.md:11]:
```suggestion
- `quantitySelectorStep` prop to `product-quantity` and `product-summary-quantity`. Enables users to control how each increment or decrement in quantity caused by these components is actually processed.
```

File: docs/README.md
Comment (victorhmp) [inline docs/README.md:70]:
```suggestion
| `quantitySelectorStep` | `enum` | Defines how the number of products that have unitMultiplier will show. Possible values are: `singleUnit` (the quantity will be not affected with the unitMultiplier) and `unitMultiplier` (the quantity will be affected with the unitMultiplier). | `unitMultiplier` |

## Customization
```

Comment (victorhmp) [reply docs/README.md:70]: It seems strange to me to describe this prop in terms of visual changes. This is more of a functional change, not a visual one.

I would wait for a review from our @vtex-apps/technical-writers :)

File: react/components/BaseProductQuantity.tsx
Comment (victorhmp) [inline react/components/BaseProductQuantity.tsx:64]:
```suggestion
    quantitySelectorStep === 'singleUnit' ? 1 : selectedItem.unitMultiplier
    
```

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: docs/README.md
Comment (germanBonacchi) [reply docs/README.md:70]: I think it's good that they review it! In case the information is useful to you, we had to develop this improvement for a client who had it as a requirement. We first tested if it generated problems with the minicart and with the checkout, and there were no problems.

File: docs/README.md
Comment (Mariana-Caetano) [inline docs/README.md:69]: Great work, @germanBonacchi! When it says:

> Defines how the number of products that have unitMultiplier will works. Possible values are (...)

That means, could we have other values, right? Do you know other possible values for this scenario and if they would have a major effect on it?

Comment (germanBonacchi) [reply docs/README.md:69]: Mmm the recommendation of make this prop an `enum` was from @victorhmp, I don't think there are other possible values.

**Review by Mariana-Caetano — APPROVED** [review]

**Review by icazevedo — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]

**Pull request merged**
